apps/src/redux/instructions.js

```diff
@@ -130,22 +130,14 @@ export default function reducer(state = {...instructionsInitialState}, action) {
     });
   }
 
-  if (
-    action.type === SET_INSTRUCTIONS_RENDERED_HEIGHT &&
-    state.allowResize &&
-    Math.abs(action.height - state.renderedHeight) > 1
```

Comment:
I know this was to work around an issue with infinite loop of resize messages, but the downside was that a single pixel adjustment wasn't possible, and with the other changes in this PR, it could lead to an inability to dismiss the scroll buttons if the instructions' height were 1 pixel away from being at full size.  In a typical resize operation, we'd get stuck with this final pixel still to go.

Reply:
cc @davidsbailey, @dmcavoy, @jmkulwik.  I really don't want to reopen the issue addressed with https://github.com/code-dot-org/code-dot-org/pull/31222 but I wasn't able to repro it locally when undoing that change.  Do any of you recall further details of a consistent repro?

Reply:
wow, nice job tracking this down! I never would have guessed that preventing a 1px resize would break something 🤷 

I have no specific recollection, but definitely worth trying tweaking all the variables listed in the slack thread including screen resolution.

no objections to potentially exposing the original issue, given the lack of repro. one idea would be to make sure Josh and PMs are aware of the original issue, since it was so hard to track down -- the user reported a complete freeze of the screen, and the workaround is (IIRC) to zoom the browser in to 100% or more.

the behavior on this page looks like a slight regression:

https://user-images.githubusercontent.com/8001765/169931571-a4ee5df1-3fd9-41b1-b95f-2c70dbb854a6.mov

this new behavior is also a little weird:

https://user-images.githubusercontent.com/8001765/169931103-715257e4-48e7-4b3d-a491-8551843aef3a.mov

not sure if these are worth addressing, happy to let y'all decide that :-)

Reply:
Dave, thanks so much for trying this locally, it's enormously helpful.  These are both great catches.  It turns out that both of these also had significant issues prior to this PR, but I've now (hopefully) made some improvements to each.

Firstly, the new scroll button logic has been extended to and customised for Minecraft.  We won't show the buttons if there's not enough room, we'll show them side by side if that's all there is room for, we'll show them vertically if there is room, and we'll hide them if there's nothing to scroll:

https://user-images.githubusercontent.com/2205926/170178306-15ef9cf0-b151-4e8a-ba41-7da6df5ed97a.mov

A lit of the jittering until now seemed to be caused by inconsistency between the height adjustment logic in `InstructionsCSF`'s `componentDidUpdate` and `UNSAFE_componentWillUpdate`, and it appears to be solved by unifying that logic in a new `adjustRenderedHeight` function that is used by both:

https://user-images.githubusercontent.com/2205926/170178542-909b3394-653b-48a1-838a-e26ec3a3dd62.mov

The other level also seems to be stable when attempting to resize now, with nowhere to go in either direction:

https://user-images.githubusercontent.com/2205926/170178581-f740e4ce-706a-48ec-a6a1-8a47e1862e59.mov

Reply:
no problem Brendan! Thanks for updating, the new videos look much better! There is still a bit of play in csf-instructions-mc-aquatic-after-normal.mov  that probably shouldn't be there, but it does not seem like a ship blocker to me as the overall scrolling experience is much improved.

Reply:
Thanks Dave.  Some extra details: 
- I think that remaining movement is due to the difference between the minimum height (established by the avatar to the left of the instructions, of all things) and the maximum height (established by the full height of the instructions text) and is expected.  Though we could revisit this with more significant changes later.
- Interestingly, and mainly as a note for posterity, is that when there is no collapse button, like in the third video, we default to "collapsed", which means that we show the full instructions and don't let you shrink from there.  I am still wondering whether this is the best behaviour.